packages/svelte/src/compiler/state.js

```diff
@@ -50,7 +53,7 @@ export function pop_ignore() {
 
 /**
  * @param {string} _source
- * @param {{ filename?: string, rootDir?: string }} options
+ * @param {{ filename?: string, rootDir?: string, filterWarning?: CompileOptions['filterWarning'] }} options
```

Comment:
What if this was required? The `parse` API would need to supply a default `() => true` option, but could also take an option which would be useful. The other place `reset` is called is in `migrate`, where we probably want to use `() => false` since there's not a lot that can be done about warnings there.

```suggestion
 * @param {{ filename?: string, rootDir?: string, filterWarning: CompileOptions['filterWarning'] }} options
```

Reply:
The other places don't care about warnings (there's nothing done with them), and we need to call `reset` anyway at which point any warnings are deleted, so not sure it's worth it.
We can definitely fall back to `() => true` in `state.js` though to avoid the dance in `compile_warnings.js`